### ISSUE TITLE
Run locally?

### ISSUE THREAD

**Issue by CHesketh76** [togethercomputer/MoA#10]
Was just wonder if I could swap out the api to local host?

**Comment by sammcj** [comment]
Was trying to figure this out too, surely it must be possible without a togetherai api key?

**Comment by sammcj** [comment]
Looks like it's currently hardcoded to use togetherai, but actually only needs any openai compatible endpoint: https://github.com/togethercomputer/MoA/blob/main/utils.py#L28

**Comment by CHesketh76** [comment]
The only method I can think of if getting something that mimics together ai's api functionality and change the endpoints in the repo.

**Comment by CHesketh76** [comment]
@sammcj that's what I was thinking as well

**Comment by sammcj** [comment]
Something like this I think https://github.com/togethercomputer/MoA/pull/12

**Comment by bolecki** [comment]
i had a quick go at this as well over here: https://github.com/togethercomputer/MoA/pull/13

it is not very configurable, but if you have an openai compatible BE hosting several models, you can just set `OPENAI_API_BASE` and update the model names in `bot.py`. i was using LM Studio and added additional models in playground. depending on your hardware, might be tough to get reasonable models to fit in VRAM.

should be easy to split out a streaming call for the aggregator as well.

**Comment by sammcj** [comment]
Got it all working - https://github.com/sammcj/MoA

**Comment by Boubou78000** [comment]
https://github.com/togethercomputer/MoA/pull/17

**Comment by yananchen1989** [comment]
@IsThatYou  hi, for the function https://github.com/togethercomputer/MoA/blob/main/utils.py#L99, in fact, it is a compatiable openai api for backend fastchat or vllm, which supports various open-llms, am I right ? It is not a true original OpenAI ?

**Comment by IsThatYou** [comment]
> @IsThatYou hi, for the function https://github.com/togethercomputer/MoA/blob/main/utils.py#L99, in fact, it is a compatiable openai api for backend fastchat or vllm, which supports various open-llms, am I right ? It is not a true original OpenAI ?

That's correct. We haven't tested with other APIs but it should be good with most openai compatible api.
